pkg/controllers/deprovisioning/controller.go

```diff
@@ -107,15 +110,22 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) controller.Bu
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	// this won't catch if the reconcile loop hangs forever but it will catch other issues
+	c.logAbnormalRuns(ctx)
+	defer c.logAbnormalRuns(ctx)
+	c.recordRun("deprovisioning-loop")
+
 	// We need to ensure that our internal cluster state mechanism is synced before we proceed
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
+		logging.FromContext(ctx).Debugf("waiting on cluster sync")
```

Comment:
We need this log line in the provisioning flow, too. I played with a PR that encapsulated this inside of cluster.Synced(ctx)

Reply:
Just added the same log statement there for now, it shouldn't be excessive as we only run that when we are scheduling a batch of pods.